### PR TITLE
fix: improve schema validation utils

### PR DESCRIPTION
**Pull request by StevePole**
## 📎 Jira ticket
n/a

## ❓ Context  
Improves JSON schema validation to better meet spec and cover additional edge cases

## 🚀 Changes 
JSON schema validation now expects 'minimum' and 'maximum', not 'min' and 'max' as per the specification. 

isNumber now return false for NaN

We reuse the our type checkers in more places to better cover edge cases.

## 💬 Considerations 
n/a

## ✅ Checklist

- [x] All changes are covered by tests
- [ ] All changes have been crossbrowser checked, especially IE11
- [ ] The changes are covered in docs

### CODE REVIEW

File: packages/dynamic-flows/src/common/validation/type-validators/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ function isString(value) {
   return typeof value === 'string';
 }
 function isNumber(value) {
-  return typeof value === 'number';
+  return typeof value === 'number' && !isNaN(value); // eslint-disable-line
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/common/validation/type-validators/index.js:5]: Is there a reason to use `isNaN` over `Number.isNaN` here? I think since you're checking typeof `Number.isNaN` is equivalent, and you can remove the warning.

File: packages/dynamic-flows/src/jsonSchemaForm/validation/schema-validators/index.js [hunk 2]
```diff
@@ -121,3 +121,5 @@ export function isValidSchema(value, schema) {
       return false;
   }
 }
+
+export { isValidSchema }; // eslint-disable-line
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/jsonSchemaForm/validation/schema-validators/index.js:125]: You shouldn't need this eslint disable I don't think, what error were you getting?

File: packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js [hunk 3]
```diff
@@ -52,6 +52,30 @@ describe('Given a library for returning the valid parts of a model based on a sc
...
+      it('should return the original integer', () => {
+        expect(result).toBe(12345);
+      });
+    });
+
+    describe('with any non integer model', () => {
+      beforeEach(() => {
+        result = getValidModelParts(12.34, schema);
+      });
+      it('should return undefined', () => {
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js:73]: should return null

File: packages/dynamic-flows/src/common/validation/type-validators/index.js (hunk 1 above)
Comment (StevePole) [reply packages/dynamic-flows/src/common/validation/type-validators/index.js:5]: No reason, good point 👍

File: packages/dynamic-flows/src/jsonSchemaForm/validation/schema-validators/index.js (hunk 2 above)
Comment (StevePole) [reply packages/dynamic-flows/src/jsonSchemaForm/validation/schema-validators/index.js]: Unfortunately this same code also exists in the angular implementation, and for some reason I needed it there.  Works fine here without 👍
